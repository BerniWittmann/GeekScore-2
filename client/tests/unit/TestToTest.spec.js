import TestToTest from "@/components/TestToTest"
import { mount } from '@vue/test-utils'

describe('TestToTest', () => {

    // Markup output tests

    it('If items are empty - hide ul', async () => {
        const wrapper = mount(TestToTest)
        wrapper.setData({
            items: []
        })
        await wrapper.vm.$nextTick()
        expect(wrapper.find('ul').isVisible()).toBe(false)
    })

    it('If items is set - show ul', async () => {
        const wrapper = mount(TestToTest)
        wrapper.setData({
            items: ['hi']
        })
        await wrapper.vm.$nextTick()
        expect(wrapper.find('ul').isVisible()).toBe(true)
    })

    it('If items is set - li contains item', async () => {
        const wrapper = mount(TestToTest)
        wrapper.setData({
            items: ['hi']
        })
        await wrapper.vm.$nextTick()
        expect(wrapper.html()).toContain('<li>hi</li>')
    })

    it('If items contain hi - li contains text hi', async () => {
        const wrapper = mount(TestToTest)
        wrapper.setData({
            items: ['hi']
        })
        await wrapper.vm.$nextTick()
        expect(wrapper.find('li').element.textContent).toBe('hi')
    })

    // Snapshot testing

    it('Snapshot test', async () => {
        const wrapper = mount(TestToTest)
        wrapper.setData({
            items: ['hi']
        })
        expect(wrapper).toMatchSnapshot()
    })


    it('Snapshot test', async () => {
        const wrapper = mount(TestToTest, {
            propsData: {
                value: 'blabla'
            }
        })
        expect(wrapper).toMatchSnapshot()
    })



    // Event tests


    it('If button is clicked - ul is hidden', async () => {
        const wrapper = mount(TestToTest)
        wrapper.find('button[type="reset"]').trigger('click')
        await wrapper.vm.$nextTick()
        expect(wrapper.find('ul').isVisible()).toBe(false)
    })

})
