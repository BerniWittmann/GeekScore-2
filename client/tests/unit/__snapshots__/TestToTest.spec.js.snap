// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestToTest Snapshot test 1`] = `
<div>
  <ul>
    <li>o</li>
  </ul> <button type="reset">Reset</button>
  <form><input data-testid="name" type="text"> <button type="submit">Get name</button></form>
</div>
`;

exports[`TestToTest Snapshot test 2`] = `
<div>
  <ul>
    <li>o</li>
  </ul> <button type="reset">Reset</button>
  <form><input data-testid="name" type="text"> <button type="submit">Get name</button></form>
</div>
`;
